### ISSUE TITLE
Execute-ProcessAsUser vbs wrapper powershell command bug

### ISSUE THREAD

**Issue by brendanmeyer** [PSAppDeployToolkit/PSAppDeployToolkit#250]
Hi

**Please be aware the Issue tracker - I can't correctly escape the code lines**

I have noticed a bug where if I am using:
`Execute-ProcessAsUser -Path "powershell.exe" -Parameters "-command ``"Start-Process -NoNewWindow '<path to program>'``""`
In which has internal  double quotes, the resulting vbs file, fails to execute
Output:
`chr(34) & "powershell.exe" & chr(34) & " -command " & chr(34) & "Start-Process -NoNewWindow '<path to program>'""`
\
basically it isn't adding the closing " & chr(34)  to the command correctly

the line that is causing the issue is:
`			[string]$executeProcessAsUserParametersVBS = 'chr(34) & ' + "`"$($Path)`"" + ' & chr(34) & ' + '" ' + ($Parameters -replace '"', "`" & chr(34) & `"" -replace ' & chr\(34\) & "$','') + '"'`

**Comment by Ernst-Mathys** [comment]
The correction to this issue is:

Change line #4624 in AppDeployToolkitMain.ps1 to

`[string]$executeProcessAsUserParametersVBS = 'chr(34) & ' + "`"$($Path)`"" + ' & chr(34) & ' + '" ' + ($Parameters -replace '"', "`" & chr(34) & `"" -replace ' & chr\(34\) & "$','') + ' & chr(34)'`

i.e. replace the double quote within the last single quotes at the end of the statement with _**& chr(34)**_

**Comment by serancti-nurun** [comment]
See the closed post # 174 and 215.

**Comment by DennisVH** [comment]
I have the same issue when using escaped or double double-quotes in Execute-ProcessAsUser Parameters
`Execute-ProcessAsUser_EY -Path "powershell.exe" -Parameters "-ExecutionPolicy Bypass -WindowStyle Hidden -File ""$dirSupportFiles\PreStageRegistryKeys.ps1""" -Wait`

**Comment by amotaz** [comment]
Can anyone else verify Ernst's suggestion. I'm going to do my own verification, however, this issue comes up repeatedly, so we need a broader test bed to confirm.

**Comment by amotaz** [comment]
Also, Dennis, can you please open another issue with the problem you mention in your comment. If you can provide some code snippets / log snippets that would be helpful as well.

**Comment by DennisVH** [comment]
amotaz, my comment wasn't clear enough. I had the same issue as brendanmeyer and using Ernst-Mathys suggestion resolved the issue.
I did some deeper digging and to reproduce the issue the Parameters value must end with a double quote:  escaped " or "" then Ernst-Mathys suggestion is the solution.
When the Parameters value does not end with a double quote the current configuration works and Ernst-Mathys solution creates a vbscript that fails.
So I suggest to use the following code:
`If ($($Parameters.Substring($Parameters.Length - 1)) -eq '"')`
`{`
`[string]$executeProcessAsUserParametersVBS = 'chr(34) & ' + "`"$($Path)`"" + ' & chr(34) & ' + '" ' + ($Parameters -replace '"', "`" & chr(34) & `"" -replace ' & chr\(34\) & "$', '') + ' & chr(34)'`
`}`
`Else`
`{`
`[string]$executeProcessAsUserParametersVBS = 'chr(34) & ' + "`"$($Path)`"" + ' & chr(34) & ' + '" ' + ($Parameters -replace '"', "`" & chr(34) & `"" -replace ' & chr\(34\) & "$', '') + '"'`
`}

double quote at the end `+ ' & chr(34)'`
not double quote at the end `+ '"'`

**Comment by DennisVH** [comment]
This will be fixed in the next release

**Comment by seanlillis** [comment]
This is fixed in v3.7.0.
